functional_tests/steps/bundles.py

```diff
@@ -121,3 +126,166 @@ def the_user_can_see_the_bundle_details_with_creator(context: Context) -> None:
+                )
+            )
+
+
+@given("there are {no_release_calendar} release calendar pages")
+def multiple_release_calendar(context: Context, no_release_calendar: str) -> None:
+    context.release_calendar_pages = []
+    if no_release_calendar.isdigit():
+        for release_calendar_index in range(int(no_release_calendar)):
+            nowish = timezone.now() + timedelta(minutes=5 * (release_calendar_index + 1))
+            context.release_calendar_pages.append(
+                ReleaseCalendarPageFactory(
+                    release_date=nowish,
+                    title="Release Calendar Page" + str(release_calendar_index),
+                    status=ReleaseStatus.CONFIRMED,
+                )
+            )
+
+
+@given("there are {no_statistical_analysis} Statistical Analysis pages")
+def multiple_statistical_analysis(context: Context, no_statistical_analysis: str) -> None:
+    context.statistical_article_pages = []
+    if no_statistical_analysis.isdigit():
+        for statistical_analysis_index in range(int(no_statistical_analysis)):
+            context.statistical_article_pages.append(
+                StatisticalArticlePageFactory(
+                    parent=ArticleSeriesPageFactory(title="PSF" + str(statistical_analysis_index))
+                )
+            )
+
+
+@given("there is a {user_role} user")
+def creat_user_by_role(context: Context, user_role: str) -> None:
+    context.user_data = create_user(user_role)
+
+
+@given("the {user_role} is a member of the Preview teams")
+def add_user_to_preview_teams(context: Context) -> None:
+    for team in context.teams:
+        user = context.user_data["user"]
+        user.teams.add(team)
+
+
+# Bundles UI Triggers
+@when("the {user_role} logs in")
+def log_in_user_by_role(context: Context) -> None:
+
+    context.page.goto(f"{context.base_url}/admin/login/")
+    context.page.get_by_placeholder("Enter your username").fill(context.user_data["username"])
+    context.page.get_by_placeholder("Enter password").fill(context.user_data["password"])
+    context.page.get_by_role("button", name="Sign in").click()
+
+
+@then("the user cannot create a bundle")
+def the_user_cannot_add_bundles(context: Context) -> None:
+    expect(context.page.get_by_role("link", name="Add bundle")).not_to_be_visible()
+    expect(context.page.get_by_text("There are no bundles to")).to_be_visible()
+
+
+@then("the user can create a bundle")
+def add_bundle_details(context: Context) -> None:
+    search_bundle = context.bundles[0]
+    context.page.get_by_role("link", name="Add bundle").click()
+    context.page.get_by_role("textbox", name="Name*").fill(search_bundle.name)
+    context.page.get_by_role("button", name="Create").click()
+
+
+@step('the user can search for unknown bundle with response "{response}"')
+def search_for_new_bundle(context: Context, response: str) -> None:
+    unknown_bundle = "Unknown bundle"
+    if unknown_bundle not in get_bundles_names(context.bundles):
```

Comment:
I don't think we should have an `if` statement like this here. This step will pass if the "Unknown bundle" is amongst the bundles in the context, without really doing what it is supposed to (i.e. check if the user can search and can see the adequate response).